bundles/org.openhab.core.automation.module.script/src/main/java/org/openhab/core/automation/module/script/ScriptTransformationService.java

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.automation.module.script;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.script.Compilable;
+import javax.script.CompiledScript;
+import javax.script.ScriptContext;
+import javax.script.ScriptEngine;
+import javax.script.ScriptException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.common.registry.RegistryChangeListener;
+import org.openhab.core.transform.TransformationConfiguration;
+import org.openhab.core.transform.TransformationConfigurationRegistry;
+import org.openhab.core.transform.TransformationException;
+import org.openhab.core.transform.TransformationService;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * The {@link ScriptTransformationService} implements a {@link TransformationService} using any available script
+ * language
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@Component(service = TransformationService.class, property = { "openhab.transform=SCRIPT" })
+@NonNullByDefault
+public class ScriptTransformationService
+        implements TransformationService, RegistryChangeListener<TransformationConfiguration> {
+    public static final String OPENHAB_TRANSFORMATION_SCRIPT = "openhab-transformation-script-";
+
+    private final Map<String, ScriptEngineContainer> scriptEngineContainers = new HashMap<>();
+    private final Map<String, CompiledScript> compiledScripts = new HashMap<>();
+    private final Map<String, String> scriptCache = new HashMap<>();
+
+    private final TransformationConfigurationRegistry transformationConfigurationRegistry;
+    private final ScriptEngineManager scriptEngineManager;
+
+    @Activate
+    public ScriptTransformationService(
+            @Reference TransformationConfigurationRegistry transformationConfigurationRegistry,
+            @Reference ScriptEngineManager scriptEngineManager) {
+        this.transformationConfigurationRegistry = transformationConfigurationRegistry;
+        this.scriptEngineManager = scriptEngineManager;
```

Comment:
I think `transformationConfigurationRegistry.addRegistryChangeListener(this)` is missing here. This causes the issue that scripts are not recompiled when they change.

Similarly `transformationConfigurationRegistry.removeRegistryChangeListener(this)` should be called on deactivation